File: bindings/kotlin/examples/CoinBalances.kt
// Copyright (c) 2025 IOTA Stiftung
// SPDX-License-Identifier: Apache-2.0

import iota_sdk.Address
import iota_sdk.GraphQlClient
import kotlinx.coroutines.runBlocking

fun main() = runBlocking {
    try {
        val client = GraphQlClient.newTestnet()
        val address =
            Address.fromHex("0xda1820edf693ee32b5729907b9b2ec8e64980ee8c008c17e89cfb4e5ecd72151")

        val coins = client.coins(address)
        for (coin in coins.data) {
            println(
                "Coin = 0x${coin.id().toHex()}, Coin Type = ${coin.coinType().asStructTag()}, Balance = ${coin.balance()}"
            )
        }

        val balance = client.balance(address) ?: 0uL
        println("Total Balance = $balance")
    } catch (e: Exception) {
        e.printStackTrace()
        kotlin.system.exitProcess(1)
    }
}

File: bindings/swift/examples/coin_balances.swift
// Copyright (c) 2026 IOTA Stiftung
// SPDX-License-Identifier: Apache-2.0

import IotaSDK

@main
struct CoinBalancesExample {
  static func main() async throws {
    let client = GraphQlClient.newDevnet()

    let address = try Address.fromHex(
      hex: "0xb14f13f5343641e5b52d144fd6f106a7058efe2f1ad44598df5cda73acf0101f")

    let coins = try await client.coins(owner: address)
    for coin in coins.data {
      print(
        "Coin = \(coin.id().toHex()), Coin Type = \(coin.coinType().asStructTag()), Balance = \(coin.balance())"
      )
    }

    let balance = try await client.balance(address: address)
    print("Total Balance = \(String(describing: balance))")
  }
}
